lib/plugins/aws/provider.js

```diff
@@ -1592,6 +1593,26 @@ class AwsProvider {
     return `${provider.deploymentPrefix}`;
   }
 
+  initS3DirectoryPaths(provider) {
+    const date = new Date();
+    const serviceStage = `${this.serverless.service.service}/${this.getStage()}`;
+    const dateString = `${date.getTime().toString()}-${date.toISOString()}`;
+    const prefix = this.provider.getDeploymentPrefix();
+
+    provider.s3CodeArtifactsDirectoryPath = `${prefix}/${serviceStage}/code-artifacts`;
+    provider.s3DeploymentDirectoryPath = `${prefix}/${serviceStage}/${dateString}`;
+  }
+
+  get s3CodeArtifactsDirectoryPath() {
```

Comment:
In context of this PR we should not yet introduce `s3CodeArtifactsDirectoryPath`, as so far path is same as `s3DeploymentDirectoryPath`.